app/scripts/views/confirm.js

```diff
@@ -115,15 +121,22 @@ define(function (require, exports, module) {
               } else {
                 self.navigate('signup_complete');
               }
-            }, function (err) {
-              // The user's email may have bounced because it was invalid.
-              // Redirect them to the sign up page with an error notice.
-              if (AuthErrors.is(err, 'SIGNUP_EMAIL_BOUNCE')) {
-                self._bouncedEmailSignup();
-              } else {
-                self.displayError(err);
-              }
             });
+        }, function (err) {
+          // The user's email may have bounced because it was invalid.
+          // Redirect them to the sign up page with an error notice.
+          if (AuthErrors.is(err, 'SIGNUP_EMAIL_BOUNCE')) {
+            self._bouncedEmailSignup();
+          } else {
+            // Hide the error from the user, an error can happen
+            // here if the status api is overloaded or if the user is switching networks.
+            // Report errors to Sentry, but not the user.
+            // Details: github.com/mozilla/fxa-content-server/issues/2638.
+            self.sentryMetrics.captureException(err);
+            self.setTimeout(function () {
+              self._startPolling();
```

Comment:
Looking at the [API docs for /recovery_email/status](https://github.com/mozilla/fxa-auth-server/blob/master/docs/api.md#get-v1recovery_emailstatus), the following error codes are possible:


* status code 401, errno 109: invalid request signature
* status code 401, errno 110: invalid authentication token
* status code 401, errno 111: invalid authentication timestamp
* status code 401, errno 115: invalid authentication nonce

Re-starting the poll isn't going to help for any of these (I believe). Perhaps for all four of these, we should send the user to the /signup page with the email pre-filled and some generic message like "There was an error signing up, sorry, please try again", or maybe we could take that a step further and automatically attempt to re-sign up the user.

Note, if the server sends a 104 and the account uid no longer exists, [we convert the error to a SIGNUP_EMAIL_BOUNCE](https://github.com/mozilla/fxa-content-server/blob/5868e4fcec23dcc26140eb9329d8e2de1a5fb8b9/app/scripts/lib/fxa-client.js#L466) locally.